api/types/kustomization_test.go

```diff
@@ -15,6 +16,12 @@ func fixKustomizationPostUnmarshallingCheck(k, e *Kustomization) bool {
 func TestFixKustomizationPostUnmarshalling(t *testing.T) {
 	var k Kustomization
 	k.Bases = append(k.Bases, "foo")
+	k.ConfigMapGenerator = []ConfigMapArgs{{GeneratorArgs{
+		KvPairSources: KvPairSources{
+			EnvSources: []string{"a", "b"},
+			EnvSource:  "c",
+		},
+	}}}
 	k.FixKustomizationPostUnmarshalling()
```

Comment:
Suggestions: 
- Should we update `fixKustomizationPostUnmarshallingCheck` to verify that `EnvSource` (singular) has been cleared?
- Both this test and the one in kusty/configmaps_test show the fix working for configmap generator. Should we add coverage for secret generator as well?

Reply:
great comments

- re: `fixKustomizationPostUnmarshallingCheck`;   Since this PR added a deepequal reflection test,  fixKustomizationPostUnmarshallingCheck is arguably a redundant and weaker condition.     I left it in so that it didn't appear that the PR was reducing coverage.  Probably should have deleted it, or used it instead of deepequals.

- one cannot see it unless one expands all the lines, but the modified krusty test does cover secret behavior - see
https://github.com/kubernetes-sigs/kustomize/pull/3692/files#diff-a497b67d0a86cac8b63450bff25432f185a5bab7413ba04f67212a7d36241cffL146
https://github.com/kubernetes-sigs/kustomize/pull/3692/files#diff-a497b67d0a86cac8b63450bff25432f185a5bab7413ba04f67212a7d36241cffR219